pyanaconda/modules/base.py

```diff
@@ -88,71 +70,92 @@ def unpublish(self):
+    @property
+    def published_tasks(self):
+        """Returns a list of published tasks."""
+        return self._published_tasks
 
-    def KickstartCommands(self) -> List[Str]:
-        """Return names of kickstart commands handled by module.
+    def publish_task(self, implementation, module_path):
+        """Publish a task."""
+        published = publish_task(implementation, module_path)
+        self._published_tasks.append(published)
 
-        :returns: List of names of kickstart commands handled by module.
-        """
-        return self.kickstart_commands
+    @property
+    def kickstart_specification(self):
+        """Return a kickstart specification.
 
-    def KickstartSections(self) -> List[Str]:
-        """Return names of kickstart sections handled by module.
+        Every kickstart module that is interested in processing
+        kickstart files, should provide its own specification.
 
-        :returns: List of names of kickstart sections handled by module.
+        :return: a subclass of KickstartSpecification
         """
-        return self.kickstart_sections
+        return NoKickstartSpecification
 
-    def KickstartAddons(self) -> List[Str]:
-        """Return names of kickstart addons handled by module.
+    @property
+    def kickstart_command_names(self):
+        """Return a list of kickstart command names."""
+        return list(self.kickstart_specification.commands.keys())
+
+    @property
+    def kickstart_section_names(self):
+        """Return a list of kickstart section names."""
+        return list(self.kickstart_specification.sections.keys())
 
-        :returns: List of names of kickstart addons handled by module.
+    @property
+    def kickstart_addon_names(self):
+        """Return a list of kickstart addon names."""
+        # TODO: We need to add support for addons.
+        return list()
+
+    def read_kickstart(self, s):
+        """Read the given kickstart string.
+
+        The kickstart string should contain only commands and
+        sections that are defined by the kickstart specification.
+
+        :param s: a kickstart string
+        :raises: instances of KickstartError
         """
-        return self.kickstart_addons
+        spec = self.kickstart_specification
+        handler = get_kickstart_handler(spec)
+        parser = get_kickstart_parser(handler, spec)
+
+        parser.readKickstartFromString(s)
+        self.process_kickstart(handler)
 
-    def ConfigureWithKickstart(self, kickstart: Str) -> Tuple[Int, Str]:
-        """Configure module with kickstart.
+    def process_kickstart(self, data):
+        """Process the kickstart data.
 
-        :returns: Tuple (Error line number, Error message) in case of parsing
-                  error or (0, "") in case of success.
+        :param data: a kickstart handler defined by the specification
         """
-        return self.configure_with_kickstart(kickstart)
+        pass
+
+    def write_kickstart(self):
```

Comment:
If it returns kickstart then it is not write. Please rename this method.

Reply:
Any suggestions?

Reply:
Maybe dump_kickstart.

Reply:
generate_kickstart ? That's what it is doing IMHO.